test/pummel/test-net-throttle.js

```diff
@@ -23,44 +23,55 @@
 require('../common');
 const assert = require('assert');
 const net = require('net');
+const debuglog = require('util').debuglog('test');
 
-const N = 1024 * 1024;
-const part_N = N / 3;
 let chars_recved = 0;
 let npauses = 0;
-
-console.log('build big string');
-const body = 'C'.repeat(N);
+let totalLength = 0;
 
 const server = net.createServer((connection) => {
-  connection.write(body.slice(0, part_N));
-  connection.write(body.slice(part_N, 2 * part_N));
-  assert.strictEqual(connection.write(body.slice(2 * part_N, N)), false);
-  console.log(`bufferSize: ${connection.bufferSize}`, 'expecting', N);
-  assert.ok(connection.bufferSize >= 0 &&
-            connection.writableLength <= N);
+  const body = 'C'.repeat(1024);
+  let keepGoing = true;
+  let n = 0;
+  debuglog('starting write loop');
+  while (keepGoing) {
```

Comment:
Nit: can we get rid of the `keepGoing` by replacing it here with `connection.write(body)`

```suggestion
  while (connection.write(body)) {
```